contracts/Router.sol

```diff
@@ -0,0 +1,188 @@
+
+    address public gov;
+    IERC20 public tree = IERC20(TREE);
+    IERC20 public reserveToken = IERC20(DAI);
+
+    struct Campaign {
+        uint256 totalPledged;
+        uint256 numPledgers;
+        uint256 treeSold;
+        mapping (uint256 => address) pledgers;
+        mapping (address => uint256) amountsPledged;
+    }
+
+    uint256 private numCampaigns = 1;
+
+    mapping(uint256 => Campaign) private campaigns;
+ 
+    constructor(address _gov) public {
+        gov = _gov;
+    }
+
+
+    function pledge(uint256 _amount, bool max) external payable {
+        require(!Address.isContract(msg.sender), "Must pledge from EOA");
+        if (max) {_amount = reserveToken.balanceOf(msg.sender);}
+        require(_amount > 0, "Must pledge more than 0.");
+        require(reserveToken.balanceOf(msg.sender) >= _amount, "Cannot pledge more reserveToken than held.");
+        reserveToken.transferFrom(msg.sender, address(this), _amount);
+
+        Campaign storage c = campaigns[numCampaigns];
+        c.totalPledged = c.totalPledged + _amount;
+
+        uint256 pledgerId = getPledgerId(numCampaigns, msg.sender);
+        if (pledgerId == 0) {
+            // user has not pledged before
+            pledgerId = c.numPledgers++;
+            c.pledgers[pledgerId] = msg.sender;
+            c.amountsPledged[msg.sender] = _amount;
+        } else {
+            // user has pledged before, add to their total pledged
+            c.amountsPledged[msg.sender] = c.amountsPledged[msg.sender].add(_amount);
+        }
+
+        emit Pledge(numCampaigns, msg.sender, _amount);
+    }
+
+
+    function unpledge(uint256 _amount, bool max) external payable {
+        Campaign storage c = campaigns[numCampaigns];
+
+        uint256 pledgerId = getPledgerId(numCampaigns, msg.sender);
+        require(pledgerId != 0, "User has not pledged.");
+        if (max) {_amount = c.amountsPledged[msg.sender];}
+        require(_amount <= c.amountsPledged[msg.sender], "Cannot unpledge more than already pledged.");
+
+        c.totalPledged = c.totalPledged.sub(_amount);
+        c.amountsPledged[msg.sender] = c.amountsPledged.sub(_amount);
+
+        reserveToken.transfer(msg.sender, _amount);
+
+        emit Unpledge(numCampaigns, msg.sender, _amount);
+    }
+
+
+    function swapExactTokensForTokens(
+        uint amountIn,
+        uint amountOutMin,
+        address[] calldata path,
+        address to,
+        uint deadline
+    ) external override returns (uint256[] memory amounts) {
+        require(deadline >= block.timestamp, 'UniswapV2Router: EXPIRED');
+
+        Campaign storage c = campaigns[numCampaigns];
+        require(c.totalPledged >= amountIn, "Not enough DAI pledged. Rebase postponed.");
+
+        // transfer pledged reserveToken to reserve
+        reserveToken.increaseAllowance(address(this), c.totalPledged);
+        reserveToken.transfer(RESERVE, c.totalPledged);
```

Comment:
The contract is supposed to be the new Reserve (aka reserve2) so it should keep all the reserve tokens.
```suggestion
```

Reply:
My mistake was assuming this contract was more of a router than reserve... Will fix